pandas/core/frame.py

```diff
@@ -4621,9 +4624,15 @@ def drop_duplicates(
         if inplace:
             (inds,) = (-duplicated)._ndarray_values.nonzero()
             new_data = self._data.take(inds)
+
+            if ignore_index:
+                new_data = new_data.reset_index(drop=True)
```

Comment:
don't use reset_index, simply use what you did in the other PR, default_index. ``.reset_index()`` causes *another* copy here

Reply:
yeah, changed!